tst/passthrough-fuse3/winposix.c

```diff
@@ -131,21 +131,44 @@ char *realpath(const char *path, char *resolved)
     int err = 0;
     WCHAR PathBuf[PATH_MAX];
     WCHAR ResultBuf[PATH_MAX];
+    PWSTR ResultBufBgn = &ResultBuf[6];
     if (0 < MultiByteToWideChar(CP_UTF8, 0, path, -1, PathBuf, PATH_MAX))
     {
-        DWORD len = GetFullPathNameW(PathBuf, PATH_MAX, ResultBuf, 0);
+        DWORD len = GetFullPathNameW(PathBuf, PATH_MAX - 6, ResultBufBgn, 0);
         if (0 == len)
             err = GetLastError();
-        else if (PATH_MAX < len)
+        else if (PATH_MAX - 6 < len)
             err = ERROR_INVALID_PARAMETER;
-        WideCharToMultiByte(CP_UTF8, 0, ResultBuf, -1, result, PATH_MAX * 4, 0, 0);
+        else
+        {
+            WideCharToMultiByte(CP_UTF8, 0, ResultBufBgn, -1, result, PATH_MAX * 4, 0, 0);
```

Comment:
According to `man 3 realpath` the result buffer will be up to `PATH_MAX` bytes, whereas we expect it to be up to `PATH_MAX * 4`. This can be problematic especially if someone passes `resolved != 0` instead of expecting `realpath` to allocate memory for the resolved path.

I do not think we need to do anything about this as `winposix.c` is a simple POSIX layer intended to be used by the passthrough file system only. Still it might be worth adding a comment above `realpath` saying something along the lines of: `/* This realpath implementation returns a string that can be up to PATH_MAX * 4 long. */`